src/append_only_data.rs

```diff
@@ -17,29 +17,44 @@ use std::{
     ops::Range,
 };
 
+/// Published sequenced AppendOnlyData.
 pub type PubSeqAppendOnlyData = SeqAppendOnlyData<PubPermissions>;
+/// Published unsequenced AppendOnlyData.
 pub type PubUnseqAppendOnlyData = UnseqAppendOnlyData<PubPermissions>;
+/// Unpublished sequenced AppendOnlyData.
 pub type UnpubSeqAppendOnlyData = SeqAppendOnlyData<UnpubPermissions>;
+/// Unpublished unsequenced AppendOnlyData.
 pub type UnpubUnseqAppendOnlyData = UnseqAppendOnlyData<UnpubPermissions>;
+/// List of entries.
 pub type Entries = Vec<Entry>;
 
+/// User that can access AppendOnlyData.
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize, Debug)]
 pub enum User {
+    /// Any user.
     Anyone,
+    /// User identified by its public key.
     Key(PublicKey),
 }
 
+/// An action on AppendOnlyData.
 #[derive(Clone, Copy, Eq, PartialEq)]
 pub enum Action {
+    /// Read from the data.
     Read,
+    /// Append to the data.
     Append,
+    /// Manage permissions.
     ManagePermissions,
 }
 
+/// Index of some data.
 #[derive(Copy, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 pub enum Index {
-    FromStart(u64), // Absolute index
-    FromEnd(u64),   // Relative index - start counting from the end
+    /// Absolute index
```

Comment:
Missing a full stop :)
```suggestion
    /// Absolute index.
```